alacritty_terminal/src/tty/windows/conpty.rs

```diff
@@ -236,33 +268,23 @@ pub fn new<C>(config: &Config<C>, size: &SizeInfo, _window_id: Option<usize>) ->
         }
     }
 
-    let conin = EventedAnonWrite::new(conin);
-    let conout = EventedAnonRead::new(conout);
-
     let child_watcher = ChildExitWatcher::new(proc_info.hProcess).unwrap();
-    let agent = Conpty { handle: pty_handle, api };
-
-    Some(Pty {
-        handle: super::PtyHandle::Conpty(ConptyHandle::new(agent)),
-        conout: super::EventedReadablePipe::Anonymous(conout),
-        conin: super::EventedWritablePipe::Anonymous(conin),
-        read_token: 0.into(),
-        write_token: 0.into(),
-        child_event_token: 0.into(),
-        child_watcher,
-    })
+
+    Some(Pty::new(super::PtyHandle::Conpty(conpty), child_watcher))
 }
 
 // Panic with the last os error as message
 fn panic_shell_spawn() {
     panic!("Unable to spawn shell: {}", Error::last_os_error());
 }
 
-impl OnResize for ConptyHandle {
+impl OnResize for ConptyResizeHandle {
     fn on_resize(&mut self, sizeinfo: &SizeInfo) {
-        if let Some(coord) = coord_from_sizeinfo(sizeinfo) {
-            let result = unsafe { (self.api.ResizePseudoConsole)(self.handle, coord) };
-            assert_eq!(result, S_OK);
+        if !self.0.closed.load(Ordering::SeqCst) {
```

Comment:
Not checking if this is closed causes the deadlock, right? Instead of failing with an error?

It also might be easier to early return here as a gate clause. Especially if this method should ever get bigger for whatever reason that makes a lot of sense. And cutting down on excessive indentation is always pretty nice too.

Reply:
> Not checking if this is closed causes the deadlock, right? Instead of failing with an error?

I'm not sure what calling `ResizePseudoConsole` on a closed pseudoconsole might do. It might just fail with an error, or it might cause some kind of undefined behaviour. The [docs for ResizePseudoConsole`](https://docs.microsoft.com/en-us/windows/console/resizepseudoconsole) just state the parameter should be "an active pseudoconsole".

I could experiment with removing this check and just ignore any error from the `ResizePseudoConsole` call. But I think the way I've currently set it up is more defensive against the untrusted C++ code :)

Reply:
> I could experiment with removing this check and just ignore any error from the ResizePseudoConsole call.

I'd definitely like to check. If it just reports an error that the file has been closed, that might make things easier for us.

Reply:
If the pseudoconsole is closed, it appears `ResizePseudoConsole` returns `E_HANDLE`. So we can work with that.